qt/scientific_interfaces/Inelastic/BayesFitting/QuasiView.cpp

```diff
@@ -0,0 +1,355 @@
+    m_uiForm.cbPlot->addItem(QString::fromStdString(PlotType::GAMMA));
+    m_uiForm.cbPlot->addItem(QString::fromStdString(PlotType::PROB));
+  } else {
+    m_uiForm.cbPlot->addItem(QString::fromStdString(PlotType::ALL));
+    m_uiForm.cbPlot->addItem(QString::fromStdString(PlotType::AMPLITUDE));
+    m_uiForm.cbPlot->addItem(QString::fromStdString(PlotType::FWHM));
+    m_uiForm.cbPlot->addItem(QString::fromStdString(PlotType::PROB));
+  }
+}
+
+void QuasiView::notifySampleInputReady(QString const &workspaceName) {
+  m_presenter->handleSampleInputReady(workspaceName.toStdString());
+}
+
+void QuasiView::notifyResolutionInputReady(QString const &workspaceName) {
+  m_presenter->handleResolutionInputReady(workspaceName.toStdString());
+}
+
+void QuasiView::notifyFileAutoLoaded() { m_presenter->handleFileAutoLoaded(); }
+
+void QuasiView::notifyPreviewSpectrumChanged(int const value) {
+  (void)value;
+  m_presenter->handlePreviewSpectrumChanged();
+}
+
+void QuasiView::notifyPlotCurrentPreview() { m_presenter->handlePlotCurrentPreview(); }
+
+void QuasiView::notifySaveClicked() { m_presenter->handleSaveClicked(); }
+
+void QuasiView::notifyPlotClicked() { m_presenter->handlePlotClicked(); }
+
+void QuasiView::minEValueChanged(double const min) {
+  disconnect(m_dblManager, &QtDoublePropertyManager::valueChanged, this, &QuasiView::updateProperties);
+  m_dblManager->setValue(m_properties["EMin"], min);
+  connect(m_dblManager, &QtDoublePropertyManager::valueChanged, this, &QuasiView::updateProperties);
+}
+
+void QuasiView::maxEValueChanged(double const max) {
+  disconnect(m_dblManager, &QtDoublePropertyManager::valueChanged, this, &QuasiView::updateProperties);
+  m_dblManager->setValue(m_properties["EMax"], max);
+  connect(m_dblManager, &QtDoublePropertyManager::valueChanged, this, &QuasiView::updateProperties);
+}
+
+void QuasiView::updateProperties(QtProperty *prop, double const value) {
+  auto eRangeSelector = m_uiForm.ppPlot->getRangeSelector("QuasiERange");
+
+  disconnect(m_dblManager, &QtDoublePropertyManager::valueChanged, this, &QuasiView::updateProperties);
+
+  if (prop == m_properties["EMin"]) {
+    InterfaceUtils::setRangeSelectorMin(m_dblManager, m_properties["EMin"], m_properties["EMax"], eRangeSelector,
+                                        value);
+  } else if (prop == m_properties["EMax"]) {
+    InterfaceUtils::setRangeSelectorMax(m_dblManager, m_properties["EMin"], m_properties["EMax"], eRangeSelector,
+                                        value);
+  }
+
+  connect(m_dblManager, &QtDoublePropertyManager::valueChanged, this, &QuasiView::updateProperties);
+}
+
+void QuasiView::handleProgramChange(int const index) {
+  int const numberOptions = m_uiForm.cbPlot->count();
+  switch (index) {
+  case 0:
+    m_uiForm.cbPlot->setItemText(numberOptions - 2, "Prob");
+    break;
+  case 1:
+    m_uiForm.cbPlot->setItemText(numberOptions - 2, "Beta");
```

Comment:
```suggestion
    m_uiForm.cbPlot->setItemText(numberOptions - 1, "Prob");
    break;
  case 1:
    m_uiForm.cbPlot->setItemText(numberOptions - 1, "Beta");
```

Since you want to switch between `Prob` and `Beta`, I think the index is the last one, as you added `Prob` to the combo box last. 
I noticed this when I changed from Lorentzians to Exponentials and it erased `FWHM` instead of `Prob`

Reply:
Thanks for spotting this!